### PR TITLE
Leaderboard: hit the API

### PR DESCRIPTION
**Pull request by zainsadaqat**
### In this pull request, I implemented the following features:

- Generate a new **ID** by creating a new game
- Implement **Refresh** button
- Implement **Submit** button
- Use **async await** feature

### CODE REVIEW

File: src/js/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+refresh.addEventListener('click', () => {
+  scoresList();
+  leaderboard.innerHTML = '';
+});
+
+form.addEventListener('submit', (e) => {
+  postData(userInput.value, scoreInput.value);
+  e.target.reset();
+});
```
Comment (henatan99) [inline src/js/index.js]: - [x] Currently, the leaderboard emptied when I reload the page. Kindly make a line of code possibly a function call here to populate the leaderboard with API data whenever the page loads or reloads. :pray: 
- [x] OPTIONAL: YOU may want to add validation to the form if `userInput.value` and `scoreInput.value` are valid inputs and display error message if not. This is just a stretch requirement and you can do it or leave it as you wish.

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @zainsadaqat , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/post-data.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  await fetch(URL, {
+    method: 'POST',
+    body: JSON.stringify({ user, score }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
+    .then((response) => response.json())
+    .then((data) => data);
+};
```
Comment (oxenprogrammer) [inline src/js/post-data.js:16]: **Optional Changes**
- [x] Nice try using the `async await` syntactic sugar to wrap your asynchronous code. However, you're mixing it with the `then` keyword. Let us remove the `then` keyword and just use the pure `async await`.

**Review by oxenprogrammer — APPROVED** [review]
Hi @zainsadaqat , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/post-data.js (hunk 2 above)
Comment (zainsadaqat) [reply src/js/post-data.js:16]: Yeah sure. Let me remove those then to make it pure async await. Thanks for pointing out the mistake.

**Comment by zainsadaqat** [comment]
Thanks, @oxenprogrammer for approving this pull request!

**Pull request merged**